Prepare the game for game version 3810 (#6324)
FAForever/fa: changelog/3810.md

And as my motivation to contribute towards the FAForever project dwindles I've decided to proceed to step down as the lead of the game team. This created a window for contributors to step up and eventually BlackYps decided to do so. He's been contributing towards the FAForever project for several years now. He's a board member of the association. In general he has the right character and intentions. And I am looking forward to what he'll do with the role.

With kind regards,

Jip

## Balance

- (#5895) After receiving several buffs, the Seraphim Tech 3 Submarine Hunter has become too oppressive, especially in larger formations. These changes aim to bring it back in line, while preserving its identity.

  - Yathsou: T3 Submarine Hunter (XSS0304):
    - Damage: 290 --> 380
    - RateOfFire: 0.25 --> 0.22
    - MuzzleSalvoDelay: 0.7 --> 1
    - MuzzleSalvoSize: 5 --> 4
    - DPS: 363 --> 338
    - Torpedo Defense RateOfFire (x2): 0.15 --> 0.1

> - (#5874) Tuning of the tech 2 naval stage.

Reviewer: ```suggestion
- (#5874) Fine tune the balance of the Tech 2 naval stage.
```
I combinded both of our sentences.

  Mobile shields do not absorb damage from ACU explosions because the ACU explosion is an anti-snowball mechanic and it would be very unfair for Aeon and UEF T2 to ignore it.

  The structure part of static shields still takes reduced structure damage from ACU explosions.

- (#6073, #6094) Allow Mongoose to fire from transports.

- (#6104) Reduce the hit points of stationary tactical missile launchers (TML) and make them volatile.

  - HP: 1500 --> 900
  - Death Weapon
    - Damage: 750
    - AoE (UEF/Aeon/Sera): 2
    - AoE (Cybran): 3

- (#6103) Greatly increase Othuum's maneuverability and slightly increase the range of its short range guns to make it better in early T3 rushes and T3 raids.

  - Othuum: T3 Siege Tank (XSL0303):
    - TurnRate: 75 -> 90
    - MaxBrake: 2.85 -> 3.55
    - Bolters' range: 20 -> 22
    - Bolters' yaw speed: 120 -> 170
    - Thau cannon yaw speed: 90 -> 125

- (#6179) Increase Ilshavoh's max turn rate from 75 to 90 and on the spot turn rate from 45 to 90.

- (#6107) Absolver can no longer damage shields while they are disabled.

> Shields can now recharge as usual.

Reviewer: ```suggestion
  Shields can now recharge as usual.
```

- (#6235) Tone down the Tempest, as it has become too oppressive and snowballed too quickly. Most importantly, its extremely powerful depth charges now fire once every 6 seconds instead of once every 5 seconds and have a reduced range. Additionally, the amount of mass required for it to attain one level of veterancy is increased from 12k to 18k.

  - Tempest: Experimental Battleship (UAS0401):
    - Oblivion Cannon Damage: 8000 --> 10000
    - Oblivion Cannon RateOfFire: 0.1 --> 0.08
    - Harmonic Depth Charge MaxRadius: 80 --> 65
    - Harmonic Depth Charge RateOfFire: 0.2 --> 0.1667
    - Harmonic Depth Charge DPS: 420 --> 350
    - VeteranMassMult: 0.5 --> 0.75
  - Atlantis: Experimental Aircraft Carrier (UES0401):
    - TurnRate: 12 --> 18
    - VeteranMassMult: 0.5 --> 0.75

- (#6252) A bug fix reverts a 1.2 -> 1.3 second reload time increase for the following unit weapons:

>   - Megalith proton cannons, CZAR AA missiles, Pillar cannons, and Cybran cruiser AA

Reviewer: ```suggestion
  - Megalith proton cannons, CZAR AA missiles, Pillar cannons, and Cybran cruiser AA.
```

- (#6169) Shoulder pods (ACU drones) now track the focus of their parent unit more closely and will automatically switch to any engineering task their parent starts. Additionally, pods can now be properly assisted by other engineering units, and assist commands targeting them will persist when the pod attaches/detaches from the parent.

- (#6160) Show the division of players in the default scoreboard instead of rating in matchmaker games

- (#6079) Make various structures easier to place to help with base building.

  - All Tech 3 Power Generators, Tech 3 Mass Fabricators, Tech 3 Artilleries, Shield Structures, Land Factories and Air Factories:
    - MaxGroundVariation: 1.0 --> 1.1
  - All Game Enders except for the Scathis:
    - MaxGroundVariation: 1.0 --> 1.2

- (#5249) Enhance the 'filter engineers' hotkey

  The hotkey now considers SACUs and tech 3 engineers to be equivalent.

  In addition it will now also filter over factions. As an example, if you have a selection of 5 Aeon engineers and 3 UEF engineers of the same tech then you'll end up selecting the Aeon engineers and the UEF engineers will assist an Aeon engineer.

- (#5249) Enhance Hotbuild

> You can now apply orders to the mobile factory through Hotbuild while you have the primary unit selected.

Reviewer: ```suggestion
  You can now apply orders to the mobile factory through Hotbuild while you have the primary unit selected.
```

- (#6110) Fix the death weapon of a unit taking up a chunk of the dispersed veterancy

  - Volatile units (example: Paragon being killed by Mavor) will now give full veterancy credit on death, where before their own death weapon would "steal" some of the credit.

- (#6115, #6201) Fix damage calculation in the unit view UI for beam weapons with a `BeamCollisionDelay` such as the Zapper's weapon.

- (#6121) Fix UEF SACU fire rate upgrade using inaccurate fire rate value.

  - The upgrade displayed 1.82x fire rate, but in fact provided 2x fire rate because of how fire rate is rounded to game ticks.

- (#6142) Fix the Salvation being unable to fire at max range by reverting previous muzzle velocity changes.
- (#6157) Allow for clan names up to 6 bytes

  - The new limit aligns with what we some (legacy) names that still exist the database.

- (#6159) Fix pause_unit_all and unpause_unit_all hotkey behavior

  - They now work as their name would suggest, instead of exactly the opposite.

- (#6162) Fix the display of engineers' reclaim statistics in the left and right UI layouts (default is the bottom layout).

- (#6168) Allow various submarines without deck guns to surface again.

> It was possible to accidentally surface these submarines. Users would then have a tough time to make the submarines dive again.

Reviewer: ```suggestion
  It was possible to accidentally surface these submarines. Users would then have a tough time to make the submarines dive again.
```

- (#6147) Improve the comments in the water shader file.

- (#6156) Mention that `OnAdjacentTo` only works for same-army structures in the annotation comment.

- (#6173) Fix slow transparency change for the tooltips of order buttons by making them use the generic tooltip creator function.

- (#6174) Annotate and document some of the internal lobby functionality.

- (#6181) Annotate fields and functions related to bomb projectiles.

- (#6188, #6198) Create a Github workflow to automate the population of the Wiki.

- (#6206) Update map blacklist.

- (#6217) Clarify warning when Hotbuild hotkeys collide with other hotkeys.

- (#6218) Annotate auto-assist toggling in `InitialAutoMode` and `SetAutoMode`.

- (#6226) Improve the visual representation of the Exodus’ and Tempest’s depth charges by increasing their `StrategicIconSize` and `UniformScale`.

- (#6229) Improve annotation for `MATH_Lerp`.

- (#6310) Enhance the periodic logging of the session time.

> The logging now differentiates between the session time and the game time. It also prints information about the allocated memory on the heap. This is useful for debugging memory issues.

Reviewer: ```suggestion
  The logging now differentiates between the session time and the game time. It also prints information about the allocated memory on the heap. This is useful for debugging memory issues.
```

> As an example: `DEBUG: Session time: 00:35:01 Game time: 00:09:33 Heap: 288.0M / 253.2M`

Reviewer: ```suggestion
  As an example: `DEBUG: Session time: 00:35:01 Game time: 00:09:33 Heap: 288.0M / 253.2M`
```

- (#6316) Add `game/worldview.lua` function `GetMousedOverWorldView()` that returns the top-most moused-over worldview, useful for in-world cursor overlays.

### Automation

> A lot of effort has gone into automating various tasks through GitHub Actions. Automation reduced the overhead on the maintainers. This is an on going process. If you have experience with GitHub Actions and you'd like to help out then do get in touch with us through Discord. We're only getting started!

Reviewer: ```suggestion
A lot of effort has gone into automating various tasks through GitHub Actions. Automation reduced the overhead on the maintainers. This is an ongoing process. If you have experience with GitHub Actions and you'd like to help out then do get in touch with us through Discord. We're only getting started!
```

- (#6223) Create Github Action workflows to automate updating various unit viewers.

> This applies to both the SpookyDB and the UnitDB unit viewers. Whenever we update the FAF game type the unit viewers are now automatically updated too.

Reviewer: ```suggestion
  This applies to both the SpookyDB and the UnitDB unit viewers. Whenever we update the FAF game type the unit viewers are now automatically updated too.
```

> - (#6139, 6138, 6137) Generate the changelog based on changelog snippets.

Reviewer: ```suggestion
- (#6139, #6138, #6137) Generate the changelog based on changelog snippets.
```

> Writing a changelog can be a daunting task. There are many changes and a lot of them can be from weeks or months ago. We now encourage all authors of pull requests to write a snippet. Through a GitHub action workflow all these snippets are dynamically compiled into a changelog. The changelog only needs a manual review before it can be released to the community.

Reviewer: ```suggestion
  Writing a changelog can be a daunting task. There are many changes and a lot of them can be from weeks or months ago. We now encourage all authors of pull requests to write a snippet. Through a GitHub action workflow all these snippets are dynamically compiled into a changelog. The changelog only needs a manual review before it can be released to the community.
```

- (#6260) Create GitHub Action workflows to automate the deployment.

> By using a workflow we open up the capabilities to post process the deployment before releasing it to the public. As a few examples that we'd like to implement:

Reviewer: ```suggestion
  By using a workflow we open up the capabilities to post process the deployment before releasing it to the public. As a few examples that we'd like to implement:
```
